### PR TITLE
[DOC] Update existing maintaining dependencies doc

### PR DESCRIPTION
**Pull request by zurex**
## Fixes
This is the first in a series of fixes for this issue #2678 .
## Changes
* Add section `how to update git submodule`
* Add section `how to update bazel dependency`
* Remove similar part in existing part.

### CODE REVIEW

**Comment by zurex** [comment]
@marcalff  would you mind take a look and help me to review this?

**Comment by marcalff** [comment]
Thanks for looking at the doc.

Please fix various CI failures (format, spelling, markdown).

**Comment by zurex** [comment]
I believe this PR meet all the requirement now. @marcalff would you mind help me to review it when you free ?

File: docs/maintaining-dependencies.md [hunk 1]
```diff
@@ -599,3 +430,209 @@ abcd1234abcd1234abcd1234abcd1234abcd1234abcd1234abcd1234abcd1234  v1.2.4.tar.gz
...
+
+```shell
+git add third_party/opentelemetry-proto
+```
+
+## Upgrade a bazel dependency
+
+Same as git submodule, we will continue use `opentelemetry-proto` as example.
+
+All the bazel dependencies is defined in [repository.bzl](https://github.com/open-telemetry/opentelemetry-cpp/blob/main/bazel/repository.bzl)
```
Comment (marcalff) [inline docs/maintaining-dependencies.md:516]:
```suggestion
All the bazel dependencies are defined in [repository.bzl](https://github.com/open-telemetry/opentelemetry-cpp/blob/main/bazel/repository.bzl)
```

**Pull request merged**